fairyfishgui.py

```diff
@@ -373,7 +387,25 @@ def run(self):
                 move = self.process_square(button)
                 if move:
                     self.update_board(move=move)
-
+            elif button == '_nnue_':
+                if not self.engine:
+                    sg.popup_ok('Please load the engine first')
+                else:
+                    nnue_path = sg.popup_get_file('Select NNUE',file_types=(('nnue file', '*.nnue'),))
+                    if nnue_path:
+                        if not (nnue_path.split('/')[-1].startswith(self.board.state.variant) if self.board.state.variant != 'chess' else nnue_path.split('/')[-1].startswith('nn')):
```

Comment:
For safely splitting path components independent of the OS (`/` vs `\`) there is `os.path.split`, or in this case more specifically `os.path.basename`. Very minor change, but looks much cleaner.

This check anyway should not really be necessary though as the engine won't complain about "wrong" NNUE files, it will just only use them once the corresponding variant is loaded, so it is safe either way. If it is meant as a notification for the user to prevent him from accidentally providing the wrong NNUE file it would need to be more sophisticated as it currently also would prevent valid usage of NNUEs, e.g., using a makruk net for ouk (see nnueAlias).